udata/core/organization/api_fields.py

```diff
@@ -27,9 +30,31 @@
 
 from udata.core.user.api_fields import user_ref_fields  # noqa: required
 
+def check_can_access_email():
+    if request.endpoint != 'api.organization':
+        return False
+
+    org = request.view_args.get('org')
+    if org is None:
+        return False
+    
+    return OrganizationPrivatePermission(org).can()
+
+# To use on public endpoint to show the email only to editor of the org (currently working only with `api.organization` endpoint, see above)
+user_in_org_with_email_fields_if_permissions = api.inherit('UserReference', user_ref_fields, {
+    'email': fields.Raw(
+        attribute=lambda o: o.email if check_can_access_email() else None,
+        description='The user email (only present on show organization endpoint if the current user has edit permission on the org)', readonly=True),
+})
+
+# To use on private endpoints where the current user is already checked to be admin/editor of the org
+user_in_org_with_always_email_fields = api.inherit('UserReference', user_ref_fields, {
```

Comment:
Maybe having only `if_permissions` differing between both is clear enough?
We could consider using only the public one, making an unnecessary check but making code simpler.
```suggestion
user_in_org_with_email_fields = api.inherit('UserReference', user_ref_fields, {
```

Reply:
Done in 880ed6935. Much simpler!